lib/pages/preferences_page.dart

```diff
@@ -10,102 +11,80 @@ class PreferencesPage extends StatefulWidget {
+          );
+        } else {
+          return Center(
+            child: Column(
                 children: <Widget>[
                   Text("Chargement de vos préférences..."),
                   Loading()
                 ]
-              ),
-            );
-          case ConnectionState.done:
-            return ListView.builder(
-              itemBuilder: (context, index) => _buildPreferencesItem(index, snapshot),
-            );
-          default:
-            return Container();
+            ),
+          );
         }
       },
     );
   }
 
-  Widget _buildPreferencesItem(int index, AsyncSnapshot<SharedPreferences> snapshot) {
+  Widget _buildPreferencesItem(int index, PreferencesLoadedState state) {
     if (index % 2 == 0) {
       index = index ~/ 2;
       switch (index) {
         case 0:
-          return ThemeSwitcher(
-            clipper: ThemeSwitcherCircleClipper(),
-            builder: (context) {
-              bool automaticDarkTheme = snapshot.data.getBool('automatic_dark_theme') ?? false;
-              if (automaticDarkTheme) {
-                return ListTile(
-                  title: Text("Mode sombre", style: Theme.of(context).textTheme.bodyText2),
-                  leading: Icon(Icons.brightness_6, color: Theme.of(context).accentColor,),
-                  trailing: Text(
-                    "Mode sombre automatique activé",
-                    style: TextStyle(color: Theme.of(context).textTheme.bodyText2.color, fontSize: 10),
-                  ),
-                );
-              } else {
-                return SwitchListTile(
-                  title: Text("Mode sombre", style: Theme.of(context).textTheme.bodyText2),
-                  secondary: Icon(Icons.brightness_6, color: Theme.of(context).accentColor,),
-                  value: snapshot.data.getBool('dark_theme') ?? false,
-                  onChanged: (dark) {
-                    if (dark) ThemeSwitcher.of(context).changeTheme(theme: AppTheme.darkTheme());
-                    else ThemeSwitcher.of(context).changeTheme(theme: AppTheme.lightTheme());
-                    snapshot.data.setBool('dark_theme', dark);
-                  }
-                );
+          if (state.automaticDarkTheme) {
+            return ListTile(
+              title: Text("Mode sombre", style: Theme.of(context).textTheme.bodyText2),
+              leading: Icon(Icons.brightness_6, color: Theme.of(context).accentColor,),
+              trailing: Text(
+                "Mode sombre automatique activé",
+                style: TextStyle(color: Theme.of(context).textTheme.bodyText2.color, fontSize: 10),
+              ),
+            );
+          } else {
+            return SwitchListTile(
+              title: Text("Mode sombre", style: Theme.of(context).textTheme.bodyText2),
+              secondary: Icon(Icons.brightness_6, color: Theme.of(context).accentColor,),
+              value: state.darkTheme,
+              onChanged: (dark) {
```

Comment:
For flag (i.e. boolean), prefer to use `isXXX`: it has a better semantic to express that a variable is a boolean with two states
```suggestion
              onChanged: (isDark) {
```